applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -98,37 +98,17 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
 
     GeometryData::IntegrationMethod GetIntegrationMethod() const override
     {
-        if (GetGeometry().LocalSpaceDimension() == 1) {
-            switch (TNumNodes) {
-            case 2:
-            case 3:
-                return GeometryData::IntegrationMethod::GI_GAUSS_2;
-            case 4:
+        switch (this->GetGeometry().GetGeometryOrderType()) {
+        case GeometryData::Kratos_Cubic_Order:
+            if (this->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D10) {
+                return GeometryData::IntegrationMethod::GI_GAUSS_4;
+            } else {
                 return GeometryData::IntegrationMethod::GI_GAUSS_3;
```

Comment:
We could also use the ternary operator here:
```suggestion
            return this->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D10
                       ? GeometryData::IntegrationMethod::GI_GAUSS_4
                       : GeometryData::IntegrationMethod::GI_GAUSS_3;
```

Reply:
done